### PR TITLE
[IGA] Add ANurbs shortcuts to Kratos namespace

### LINKED ISSUE #3403: [IGA] IgaApplication does not compile under Linux
A lot of errors I don't know how to fix (with ANurbs Master from today):
[err.txt](https://github.com/KratosMultiphysics/Kratos/files/2592212/err.txt)

=> will be nice to have #3398 

FYI @tteschemacher

### PR DESCRIPTION
**Pull request by oberbichler**
In this branch I...
- add some shortcuts for geometric entities to the `namespace Kratos`
- use the Kratos alias for `shared_ptr`
- do some cleanup

fixes #3403

### CODE REVIEW

File: applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp
Comment (philbucher) [inline applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp]: what is this? an `std` or a `Kratos`  `shared_ptr`?
=> can you please prepend `Kratos::`? (also in other places)

File: applications/IgaApplication/custom_utilities/anurbs.h
Comment (philbucher) [inline applications/IgaApplication/custom_utilities/anurbs.h:56]: @jcotela does this need approval of the committee?

File: applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp
Comment (philbucher) [inline applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp:442]: this is easy to read and hard to read at the same time :D
Fine for me

**Review by philbucher — COMMENTED** [review]
some comments, please wait for the reply of @jcotela

File: applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp
Comment (oberbichler) [reply applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp]: Well I am already in the Kratos namespace... but with the shared_ptr it might make sense to explicit add the namespace. Otherwise some people might be confused.

File: applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp
Comment (oberbichler) [reply applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp:442]: `Holder` is from the pybind11 documentation

File: applications/IgaApplication/custom_utilities/anurbs.h
Comment (oberbichler) [reply applications/IgaApplication/custom_utilities/anurbs.h:56]: Options: give then a long name or put them into a namespace

**Comment by pooyan-dadvand** [comment]
I will take a look at it

File: applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp
Comment (philbucher) [reply applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp]: yes it would be good to add it IMO

**Review by philbucher — COMMENTED** [review]
ping @pooyan-dadvand

**Comment by pooyan-dadvand** [comment]
I would say that CurvedLine and CurvedSurface would be more descriptive....

**Comment by oberbichler** [comment]
@pooyan-dadvand very subjective... Rhino/OpenCascade e.g. are using `Curve` and `Surface`. But at the end it is just a modification of the `using`...
Putting this objects inside the Kratos-namespace is fine for you?

**Comment by pooyan-dadvand** [comment]
Yes in general I'm fine with adding them to the namespace and for this reason I added the issue to technical committee todo. Nevertheless, this list is too saturated....

**Comment by pooyan-dadvand** [comment]
> @pooyan-dadvand very subjective...

Of course! This was my personal opinion but I admit that coherence is also important!

**Comment by oberbichler** [comment]
This pull request should now also solves the problems when compiling the IgaApplication under Linux.

I would wait with the renaming of `CurveGeometry` to `NurbsCurveGeometry`  due to other pending pull requests (same with surface).

I would keep the name `Curve` because...
...same naming as in other packages.
...the underlying geometry theoretically could also be a line, polyline, etc.
...otherwise `NurbsCurveGeometry` would become `NurbsCurvedLineGeometry` which is quite long.
(But if it is important or it causes namespace issues I am also not completely against it)

If this is OK I would like to merge this pull request.

**Comment by pooyan-dadvand** [comment]
I agree with maintaining the names

**Review by philbucher — APPROVED** [review]
I am approving since from what I understood Pooyan is fine with the changes

Also this PR is blocking many other developments
Plus it fixes the linux-compilation

@KratosMultiphysics/technical-committee as @oberbichler pointed out, if you have a problem with adding to the namespace or other things, then please comment it here and we will change it in a future PR

**Pull request merged**
